sender-kinesis/src/main/java/zipkin/reporter/kinesis/KinesisSender.java

```diff
@@ -0,0 +1,158 @@
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.reporter.kinesis;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.handlers.AsyncHandler;
+import com.amazonaws.services.kinesis.AmazonKinesisAsync;
+import com.amazonaws.services.kinesis.AmazonKinesisAsyncClientBuilder;
+import com.amazonaws.services.kinesis.model.PutRecordRequest;
+import com.amazonaws.services.kinesis.model.PutRecordResult;
+import com.google.auto.value.AutoValue;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import zipkin.internal.LazyCloseable;
+import zipkin.internal.Nullable;
+import zipkin.reporter.BytesMessageEncoder;
+import zipkin.reporter.Callback;
+import zipkin.reporter.Encoding;
+import zipkin.reporter.Sender;
+
+@AutoValue
+public abstract class KinesisSender extends LazyCloseable<AmazonKinesisAsync> implements Sender {
+
+  public static KinesisSender create(String streamName) {
+    return builder().streamName(streamName).build();
+  }
+
+  public static Builder builder() {
+    return new AutoValue_KinesisSender.Builder()
+        .credentialsProvider(new DefaultAWSCredentialsProviderChain())
+        .messageMaxBytes(1024 * 1024); // 1MB Kinesis limit.
+  }
+
+  @AutoValue.Builder
+  public interface Builder {
+
+    /** Kinesis stream to send spans. */
+    Builder streamName(String streamName);
+
+    /** AWS credentials for authenticating calls to Kinesis. */
+    Builder credentialsProvider(AWSCredentialsProvider credentialsProvider);
+
+    Builder endpointConfiguration(AwsClientBuilder.EndpointConfiguration endpointConfiguration);
+
+    /** Maximum size of a message. Kinesis max message size is 1MB */
+    Builder messageMaxBytes(int messageMaxBytes);
+
+    KinesisSender build();
+  }
+
+  abstract String streamName();
+
+  @Nullable
+  abstract AWSCredentialsProvider credentialsProvider();
+
+  // Needed to be able to overwrite for tests
+  @Nullable
+  abstract AwsClientBuilder.EndpointConfiguration endpointConfiguration();
+
+  private final AtomicBoolean closeCalled = new AtomicBoolean(false);
+
+  @Override
+  public CheckResult check() {
+    try {
+      // TODO are we OK requiring DescribeStream be available to us as a publisher?
```

Comment:
I'm ok with it. We can add to the readme, and if someone complains, we can always address later.

ps this should be mock testable (the success and failure case)